core/trino-main/src/test/java/io/trino/block/BlockAssertions.java

```diff
@@ -227,6 +230,13 @@ public static Block createRandomLongDecimalsBlock(int positionCount, float nullR
                 () -> String.valueOf(RANDOM.nextLong())));
     }
 
+    public static Block createRandomLongsBlock(int positionCount, int numberOfUniqueValues)
+    {
+        checkArgument(positionCount >= numberOfUniqueValues, "numberOfUniqueValues must be between 1 and positionCount: %s but was %s", positionCount, numberOfUniqueValues);
+        int[] randomPartitions = chooseRandomUnique(positionCount, numberOfUniqueValues).toIntArray();
+        return createLongsBlock(IntStream.range(0, positionCount).mapToLong(position -> randomPartitions[RANDOM.nextInt(numberOfUniqueValues)]).boxed().collect(toImmutableList()));
```

Comment:
1. Why not just pick random value from `[0, numberOfUniqueValues)` it's as good as this solution
nit: this doesn't really guarantee that there will be really `numberOfUniqueValues` unique values.

Anyway, from processing POV long block with `numberOfUniqueValues` and all random values shouldn't be that different (apart from memory locality in joins maybe). Maybe just remove this method?

2. move `.mapToLong` `.boxed` to separate lines

Reply:
If we only pick random from  `[0, numberOfUniqueValues)` as a value and then partition by that block, we will have partitions with non zero elements only at the first `numberOfUniqueValues` positions
e.g. `[4,5,3,0,0,0,0,0,0,0]` where in the real world I expect something like `[0,0,4,5,0,0,03,0,0]`.
The first pattern is easy to predict by for example branch predictor, which may skew benchmarking results.